src/pages/workspaces/workspace/notebooks/TerminalLauncher.js

```diff
@@ -64,7 +64,8 @@ const TerminalLauncher = _.flow(
   }
 
   async componentDidUpdate() {
-    const { cluster: { clusterUrl } = {} } = this.props
+    const { cluster } = this.props || {}
```

Comment:
`this.props` is always an object, so there's no need to provide a fallback value here.